direct/src/directnotify/RotatingLog.py

```diff
@@ -8,7 +11,12 @@ class RotatingLog:
     to a new file if the prior file is too large or after a time interval.
     """
 
-    def __init__(self, path="./log_file", hourInterval=24, megabyteLimit=1024):
+    def __init__(
+        self,
+        path: str = "./log_file",
+        hourInterval: int | None = 24,
+        megabyteLimit: int | None = 1024,
+    ) -> None:
```

Comment:
Should we also add types to `self.timeInterval`, `self.timeLimit`, and `self.sizeLimit`?

Reply:
It looks like mypy and pyright are both able to correctly infer the types, as `__init__` assigns as both numeric values and `None`. So we could, but we don't need to.

Reply:
Awesome, great to see type inference working here.